### PR TITLE
Pure Java Grex with alternate naming structure

### PR DESCRIPTION
**Pull request by grodin**
This is the same as #7 but with the pure Java submodule renamed to `grex` and the Android submodule renamed to `grex-android`.

I haven't touched the Maven setup.

Edit - including comments from #7 since we'll be closing that if we merge this:

Things not updated in this PR:

-    README
-    CHANGELOG
-    No new Maven artifact, which there probably should be, not sure how that should go

### CODE REVIEW

File: grex-android/src/main/java/au/com/gridstone/grex/GRexPersister.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import static android.content.Context.MODE_PRIVATE;
+
+/**
+ * Facilitates the read and write of objects to and from an application's
+ * private directory.
+ *
+ * @author Christopher Horner
+ * @author Joseph Cooper
+ */
+public class GRexPersister extends BaseGRexPersister {
```
Comment (chris-horner) [inline grex-android/src/main/java/au/com/gridstone/grex/GRexPersister.java]: How do you feel about renaming this to GRexAndroidPersister?

**Comment by chris-horner** [comment]
I know the API break is going to suck, but I'm much more inclined to go with this structure. Going forward, it'll be nice to have a codebase that makes a little more sense.

What do you think? Is this structure worth the API break?

**Comment by grodin** [comment]
I've had a think about it and I think we should go forward with this new structure, but we should hold off on a release for a short time (say a few weeks). 

My idea is that if we're going to go and break the API, let's get in any other breaking changes we want so we can then settle the API down a bit. I've got at least one suggestion, to switch to static factory methods instead of constructors so we can return instances of the interface instead of the implementations, meaning we can change things more easily in the future.

My proposal is that instead of merging this PR into `master`, merge it into a new branch, say `api-break` or something, and then we can sit on that for a little while, use that branch as a target for breaking PRs to allow discussing any other breaking changes before releasing.

The main disadvantage with that I can foresee is that any PRs targetting `master` received in the meantime will be a major pain to merge. On the other hand, it's not like there have been loads of PRs (except from me!) recently, so that shouldn't be a problem for a couple of weeks.

What do you think of that idea?

File: grex-android/src/main/java/au/com/gridstone/grex/GRexPersister.java (hunk 1 above)
Comment (grodin) [inline grex-android/src/main/java/au/com/gridstone/grex/GRexPersister.java]: I was going to rename a few other classes too, but forgot. I'll do those ASAP.

**Comment by cristiangarciascmspain** [comment]
Don't worry so much about not breaking the API, just make another major release and put a good description of the changes on the changelog.
If the changelog and the new documentation has sense I think nobody will complain. Major releases break some interfaces.

**Comment by IgorGanapolsky** [comment]
Will this pull request ever be merged?

**Comment by chris-horner** [comment]
It will. I've just wanted to put some thought into how we go about this. Ultimately, I think the simplest approach is the best one and we'll just merge this into master and break the API.

@grodin Have you had a chance to rename `GRexPersister` to `GRexAndroidPersister`?

**Comment by grodin** [comment]
Sorry, I forgot I needed to do that. Will do it tomorrow so we can get it
merged asap.

**Comment by grodin** [comment]
Done.

**Comment by grodin** [comment]
Just realised, I may have inadvertently included some copyright notices on files that I added. I don't know what the normal practice is for stuff like that with contributed code. Let me know if anything needs changing.

**Comment by chris-horner** [comment]
Thanks so much for all your work on this, and sorry I took so long to getting around to merging it! Nice work :)

**Comment by grodin** [comment]
No worries, thanks for putting up the original thing in the first place! I'm just glad to contribute.

**Pull request merged**